tests/show_techsupport/test_techsupport.py

```diff
@@ -0,0 +1,515 @@
+
+    logger.info('generate config for MIRROR table {}'.format(everflow_table_name))
+    src = os.path.join(TEMPLATE_DIR, ACL_TABLE_TEMPLATE)
+    everflow_config = 'everflow_table_' + everflow_table_name + '.json'
+    everflow_config_path = os.path.join(tmp_dir, everflow_config)
+    duthost.template(src=src, dest=everflow_config_path)
+
+    yield {
+        'name' : ACL_TABLE_NAME,
+        'config_file' : everflow_config_path 
+    }
+
+
+
+@pytest.fixture(scope='function')
+def mirror_table(duthost, mirror_table_config):
+    """
+    generate everflow table with acl_stage=ingress, default mirror_stage=ingress
+    """
+
+    name = mirror_table_config['name']
+    conf = mirror_table_config['config_file']
+    loganalyzer = LogAnalyzer(ansible_host=duthost, marker_prefix='acl')
+    loganalyzer.load_common_config()
+
+    try:
+        loganalyzer.expect_regex = [LOG_EXCEPT_EVERFLOW_TABLE_CREATE]
+        with loganalyzer:
+            logger.info('creating MIRROR table: applying {}'.format(name))
+            duthost.command('sonic-cfggen -j {} --write-to-db'.format(conf))
+    except LogAnalyzerError as err:
+        logger.info('removing EVERFLOW table {}'.format(name))
+        duthost.command('config acl remove table {}'.format(name))
+        raise err
+    
+    try:
+        yield mirror_table_config
+    finally:
+        loganalyzer.expect_regex = [LOG_EXCEPT_EVERFLOW_TABLE_DELETE]
+
+        with loganalyzer:
+            logger.info('removing EVERFLOW table {}'.format(name))
+            duthost.command('config acl remove table {}'.format(name))
+        duthost.command('config save -y')
+
+
+@pytest.fixture(scope='function')
+def mirror_setup(duthost):
+    """
+    setup fixture
+    """
+    ports = []
+    port_channels = []
+    acl_table_ports = []
+
+    mg_facts = duthost.minigraph_facts(host=duthost.hostname)['ansible_facts']
+
+    for dut_port, neigh in mg_facts['minigraph_neighbors'].items():
+        if 'T0' in neigh['name'] or 'T2' in neigh['name']:
+            ports.append(dut_port)
+
+    # get the list of port channels
+    port_channels = mg_facts['minigraph_portchannels']
+    acl_table_ports += ports
+    acl_table_ports += port_channels
+    
+    # remove default SONiC Everflow table (allowed to have only one mirror table)
+    duthost.command('config acl remove table EVERFLOW')
+    logger.debug("creating running directory ...")
+    duthost.command('mkdir -p {}'.format(MIRROR_RUN_DIR))
+
+    setup_info = {
+        'dut_tmp_dir': MIRROR_RUN_DIR,
+        'port_channels': port_channels,
+        'acl_table_ports': acl_table_ports,
+    }
+    logger.info('setup variables {}'.format(pprint.pformat(setup_info)))
+    yield setup_info
+    
+    teardown_mirroring(duthost, mirror_setup)
+    logger.info('removing {}'.format(MIRROR_RUN_DIR))
+    duthost.command('rm -rf {}'.format(MIRROR_RUN_DIR))
+
+
+@pytest.fixture(scope='function')
+def session_info():
+    yield {
```

Comment:
This is constant, no need to have it like a seperate fixture

Reply:
made it constant dictionary